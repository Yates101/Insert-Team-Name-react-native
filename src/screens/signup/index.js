<<<<<<< HEAD
export * from './signup';
=======
export * from './signup'
>>>>>>> origin/logout
