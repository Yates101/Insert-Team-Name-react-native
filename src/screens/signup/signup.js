import React, { useState, useEffect } from 'react';
import { StyleSheet, Text, View, ScrollView, TextInput, Button } from 'react-native';
import { Headbar } from '../../components/headbar'
import axios from 'axios';

const SignUpScreen = ({ navigation }) => {

const [foreName, setForeName] = useState('')
const [surName, setSurName] = useState('')
const [userName, setUserName] = useState('')
const [email, setEmail] = useState('')
const [passWord, setPassWord] = useState('')

const createUser = async() => {
    try{
      axios.post("http://localhost:3001/users",
        { "forename": foreName, "surname": surName, "username": userName, "email": email, "password": passWord, "profilePic": "" })
        navigation.navigate("Log In");
    }
    catch(error){
      console.log('error', {error});
    }
  }

  return(
    <View>
    <Headbar/>
      <View style={styles.container}>
        <TextInput style={styles.textInput}
          placeholder="first name"
          onChangeText={(text) => setForeName(text)}
          value={foreName}
        />
        <TextInput style={styles.textInput}
          placeholder="last name"
          onChangeText={(text) => setSurName(text)}
          value={surName}
        />
        <TextInput style={styles.textInput}
          placeholder="username"
          onChangeText={(text) => setUserName(text)}
          value={userName}
        />
        <TextInput style={styles.textInput}
          placeholder="email"
          onChangeText={(text) => setEmail(text)}
          value={email}
        />
        <TextInput style={styles.textInput}
          placeholder="password"
          onChangeText={(text) => setPassWord(text)}
          value={passWord}
        />      
  
        <View style={styles.buttonContainer}>
          <Button title="sign up" style={styles.button} onPress={() => {createUser();}}/>
        </View>

      </View>
    </View>
  )            
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#696969',
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
  },

  textInput: {
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderLeftWidth: 1,
    borderRightWidth: 1,
    borderColor: '#E1E5EA',
    borderRadius: 4,
    paddingTop: 15,
    paddingBottom: 15,
    paddingLeft: 14,
    paddingRight: 14,
    fontSize: 16,
    color: 'black',
    backgroundColor: 'white',
    width: 200,
  },

  button: {
    color: 'white',
    width: 100,
  },

  buttonContainer: {
    backgroundColor: 'blue',
  }

});

export { SignUpScreen };