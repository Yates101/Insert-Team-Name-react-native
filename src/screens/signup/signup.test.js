import 'react-native';
import React from 'react';

import { SignUpScreen } from './signup.js';
import renderer from 'react-test-renderer';
import '@testing-library/jest-native/extend-expect';

describe("<SignUp />", () => {
  const tree = renderer.create(<SignUpScreen />).toJSON();

  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  })

  it('has 6 children', () => {
    expect(tree.children.length).toBe(6);
  });
})
