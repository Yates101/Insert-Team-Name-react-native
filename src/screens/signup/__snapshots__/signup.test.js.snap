// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "alignSelf": "stretch",
        "backgroundColor": "#179aff",
        "flexDirection": "row",
        "height": 52,
        "justifyContent": "space-between",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFED00",
          "fontSize": 32,
          "fontWeight": "bold",
        }
      }
    >
      IkeBook
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#cdd7d6",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="first name"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "#E1E5EA",
          "borderLeftWidth": 1,
          "borderRadius": 4,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "color": "black",
          "fontSize": 16,
          "paddingBottom": 15,
          "paddingLeft": 14,
          "paddingRight": 14,
          "paddingTop": 15,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="last name"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "#E1E5EA",
          "borderLeftWidth": 1,
          "borderRadius": 4,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "color": "black",
          "fontSize": 16,
          "paddingBottom": 15,
          "paddingLeft": 14,
          "paddingRight": 14,
          "paddingTop": 15,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="username"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "#E1E5EA",
          "borderLeftWidth": 1,
          "borderRadius": 4,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "color": "black",
          "fontSize": 16,
          "paddingBottom": 15,
          "paddingLeft": 14,
          "paddingRight": 14,
          "paddingTop": 15,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="email"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "#E1E5EA",
          "borderLeftWidth": 1,
          "borderRadius": 4,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "color": "black",
          "fontSize": 16,
          "paddingBottom": 15,
          "paddingLeft": 14,
          "paddingRight": 14,
          "paddingTop": 15,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "#E1E5EA",
          "borderLeftWidth": 1,
          "borderRadius": 4,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "color": "black",
          "fontSize": 16,
          "paddingBottom": 15,
          "paddingLeft": 14,
          "paddingRight": 14,
          "paddingTop": 15,
          "width": 200,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "backgroundColor": "blue",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            sign up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
